### PR TITLE
Create deploy-delegate-to-ECS

### PR DESCRIPTION
**Pull request by kat-enos**
Quick step-by-step on how to install an immutable delegate on Amazon ECS and AWS Fargate.

# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [x ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*: To be noted later on

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS [hunk 1]
```diff
@@ -0,0 +1,258 @@
+---
+title: Deploy a NextGen delegate to Amazon ECS
```
Comment (pratmit) [inline docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS:2]: There is no NextGen Delegate. Rephrase this to " Deploy a Delegate to Amazon ECS in Harness Nextgen".

File: docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+description: Provides information and YAML for the installation of a NextGen delegate into an Amazon ECS or AWS Fargate cluster.
+# sidebar_position: 2
+---
+
+# Deploy a NextGen delegate to Amazon ECS
```
Comment (pratmit) [inline docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS:7]: Same as above, NextGen delegate is not the right name.

File: docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+   ```
+   
+   Replace `service-name` with [TBD]. Replace `SERVICE_NAME` with [TBD]. Replace `task-definition` with [TBD].
+   
+2. Use the following instruction to increase the count of replica pods to the desired number:
+
+   ```
+   harness-delegate-task-spec --cluster <CLUSTER_NAME> --desired-count 1
+   
+## Deploy a delegate to Amazon Fargate
```
Comment (pratmit) [inline docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS:124]: Deploy a delegate to Amazon Fargate in Harness NextGen.

**Review by pratmit — CHANGES_REQUESTED** [review]

File: docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS (hunk 1 above)
Comment (kat-enos) [reply docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS:2]: I thought that was awkward ("to Amazon ECS in Harness NextGen") so tried to create a simpler title. Could also try "Use Harness NextGen to deploy a delegate to Amazon ECS".

File: docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS (hunk 2 above)
Comment (kat-enos) [reply docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS:7]: Once the context is set, it's not necessary to repeat it anyway. We could just talk about deploying a delegate and say that the process requires NextGen too. You don't usually get or want to say everything in the title -- makes it too long. Usually you set the context/requirements in the first paragraph.

Comment (kat-enos) [reply docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS:7]: But I'll integrate whatever changes, including these, tomorrow.

Comment (pratmit) [reply docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS:7]: That works for me but we can't say NextGen Delegate.

File: docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS [hunk 4]
```diff
@@ -0,0 +1,270 @@
+---
+title: Deploy a delegate to Amazon ECS or AWS Fargate
+description: Provides information and YAML for the installation of a delegate into an Amazon ECS or AWS Fargate cluster.
+# sidebar_position: 2
+---
+
+# Deploy a delegate to Amazon ECS or AWS Fargate
```
Comment (pratmit) [inline docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS:7]: Deploy a docker delegate

File: docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS (hunk 2 above)
Comment (kat-enos) [reply docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS]: Fixed.

**Review by pratmit — DISMISSED** [review]

File: docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+2. Edit the fields of the task definition as follows.
+
+   | **Field** | **Description** |
+   | :-- | :-- |
+   | `ACCOUNT_ID | Your Harness account ID. |
+   | `DELEGATE_TOKEN` | The delegate token stored in your Harness account. |
+   | `MANAGER_HOST_AND_PORT` | Information about your manager host. This depends on the Harness production cluster you use: Prod1: https://app.harness.io, Prod2: https://app.harness.io/gratis, or Prod3: https://app3.harness.io. |
+   | `DELEGATE_NAME` | The name you gave your delegate. This is usually the name you specified during delegate installation. |
+   | `LOG_STREAMING_SERVICE_URL` | The URL of your log streaming service. This depends on the Harness production cluster you use: Prod1: https://app.harness.io/log-service/, Prod2: https://app.harness.io/gratis/log-service/, or Prod3: https://app3.harness.io/log-service/. |
+
```
Comment (bharatgoelharness) [inline docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS:226]: For image also we can add a description
IMAGE: Check the newer version of delegate image from https://hub.docker.com/r/harness/delegate/tags

File: docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+2. Enter the fields of the task definition as follows:
+
+   | **Field** | **Description** |
+   | :-- | :-- |
+   | `ACCOUNT_ID | Your Harness account ID. |
+   | `DELEGATE_TOKEN` | The delegate token stored in your Harness account. |
+   | `MANAGER_HOST_AND_PORT` | Information about your manager host. This depends on the Harness production cluster you use: Prod1: https://app.harness.io, Prod2: https://app.harness.io/gratis, or Prod3: https://app3.harness.io. |
+   | `DELEGATE_NAME` | The name you gave your delegate. This is usually the name you specified during delegate installation. |
+   | `LOG_STREAMING_SERVICE_URL` | The URL of your log streaming service. This depends on the Harness production cluster you use: Prod1: https://app.harness.io/log-service/, Prod2: https://app.harness.io/gratis/log-service/, or Prod3: https://app3.harness.io/log-service/. |
+
```
Comment (bharatgoelharness) [inline docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS:107]: Here also we should add for image

Comment (kat-enos) [reply docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS:107]: Yes. Trying to fix that now.  :]
But I don't see an IMAGE env var in the YAML. Does IMAGE refer to the "image" field?

Comment (kat-enos) [reply docs/platform/2_Delegates/advanced-installation/deploy-delegate-to-ECS:108]: Now it's in both tables Bharat. Is that what you want? But there is no env var with the same IMAGE -- just the "image" field. Is that okay?

**Review by pratmit — APPROVED** [review]
LGTM

**Comment by kat-enos** [comment]
Banzai!
